src/Plugins/RestServer/Middleware/RestServerMiddleware.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// RestServerMiddleware.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Microsoft.AspNetCore.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Neo.Plugins.RestServer.Middleware
+{
+    internal class RestServerMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public RestServerMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var request = context.Request;
+            var response = context.Response;
+
+            SetServerInformationHeader(response);
+
+            await _next(context);
+        }
+
+        public static void SetServerInformationHeader(HttpResponse response)
+        {
+            var neoCliAsm = Assembly.GetEntryAssembly()?.GetName();
```

Comment:
We should allow or disallow this in the settings, and also cache the value in order to avoid reflection waste of time.